android/java/org/chromium/chrome/browser/crypto_wallet/util/ParsedTransaction.java

```diff
@@ -245,11 +246,11 @@ public static ParsedTransaction parseTransaction(TransactionInfo txInfo,
 
         final ParsedTransactionFees feeDetails = ParsedTransactionFees.parseTransactionFees(
                 txInfo, selectedNetwork, networkSpotPrice, solFeeEstimatesFee);
-
         TxDataUnion txDataUnion = txInfo.txDataUnion;
-        TxData1559 txData = txDataUnion.getEthTxData1559();
-        SolanaTxData solTxData = null; // TODO: add with SOL
-        FilTxData filTxData = null; // TODO: add with FIL
+        // TxDataUnion has some unnecessary assertion in getter methods that crashes the app
```

Comment:
remove an `unnecessary` word to avoid confusing. It doesn't mean that it is unnecessary If you are not 100% confident and not sure why that assertion was put in some particular place. In most cases you just don't know all the cases that could appear. I described you in a slack that most likely in some cases a few structures can be set to not null, but we need to get only the one that has a corresponding Tag set.

Reply:
Update though I think I explained it before but this is unnecessary because it was working fine before but now causing a crash. Personally I think assert here and in this way is an opinionated practice but not gonna remove any :)

Reply:
we used it wrong before as we used only one item from the union, but we should have checked Tag always even for 1 item. It doesn't mean we should continue to use it wrong If it's been used wrong before. It's good that you faced that problem and we didn't have strange behaviours when we released that feature.